README.md

```diff
@@ -85,6 +85,44 @@ ergonomic to write servers without dealing with sockets or serialization
 directly. Simply implement one of the generated traits, and you're off to the
 races! See the tarpc_examples package for more examples.
 
+## Example: Futures
+
+Here's the same server, implemented using `FutureService`.
+
+```rust
+#![feature(conservative_impl_trait, plugin)]
+#![plugin(tarpc_plugins)]
+
+extern crate futures;
+#[macro_use]
+extern crate tarpc;
+
+use tarpc::util::{FirstSocketAddr, Never};
+use tarpc::sync::Connect;
+
+service! {
+    rpc hello(name: String) -> String;
+}
+
+#[derive(Clone)]
+struct HelloServer;
+
+impl FutureService for HelloServer {
+    type HelloFut = futures::Finished<String, Never>;
```

Comment:
I didn't expect it to be any particular type -- I saw it was used in the return type for `hello`, and so inferred that the two were related. However, I didn't know if it was *required* that I declare the type alias, or whether I could have just put the type directly in the definition of `hello`.